javaScript/signup.js

```diff
@@ -0,0 +1,134 @@
+document.addEventListener("DOMContentLoaded", function () {
+  const emailInput = document.getElementById("email");
+  const emailError = document.getElementById("email-error");
+
+  const usernameInput = document.getElementById("username");
+  const usernameError = document.getElementById("username-error");
+
+  const passwordInput = document.getElementById("password");
+  const passwordError = document.getElementById("password-error");
+
+  const confirmPasswordInput = document.getElementById("confirm-password");
+  const confirmPasswordError = document.getElementById("confirm-password-error");
+
+  const passwordToggle = document.querySelector("#password-toggle");
+  const confirmPasswordToggle = document.querySelector("#confirm-password-toggle");
+
+  const signupButton = document.querySelector(".login-btn");
+
+  function showError(input, errorElement, message) {
+    input.style.border = "2px solid red";
+    errorElement.textContent = message;
+    errorElement.style.color = "red";
+    errorElement.style.display = "block";
+    updateSignupButtonState();
+  }
+
+  function hideError(input, errorElement) {
+    input.style.border = "";
+    errorElement.textContent = "";
+    errorElement.style.display = "none";
+    updateSignupButtonState();
+  }
+
+  function validateInput() {
+    let isValid = true;
+
+    if (emailInput.value.trim() === "" || emailError.textContent !== "") isValid = false;
+    if (usernameInput.value.trim() === "" || usernameError.textContent !== "") isValid = false;
+    if (passwordInput.value.trim() === "" || passwordError.textContent !== "") isValid = false;
+    if (confirmPasswordInput.value.trim() === "" || confirmPasswordError.textContent !== "") isValid = false;
+
+    return isValid;
+  }
```

Comment:
💊 제안
해당 함수를 보면 if문 조건에 해당하는 경우 false를 반환하니 분기를 이렇게 나눠주실 필요가 없을 것 같아요~
또한 isValid 변수는 단순히 true 또는 false 값을 저장하는 용도로만 사용되므로, 별도로 선언할 필요 없이 바로 반환하는 것이 더 효율적일 것 같아요~

```suggestion
  function validateInput() {
    if (emailInput.value.trim() === "" || emailError.textContent !== "" || usernameInput.value.trim() === "" || usernameError.textContent !== "" || passwordInput.value.trim() === "" || passwordError.textContent !== "" || confirmPasswordInput.value.trim() === "" || confirmPasswordError.textContent !== "") return false;

    return true;
  }
```

혹은 아래처럼도 가능합니다.

```suggestion
function validateInput() {
  return (
    emailInput.value.trim() !== "" && emailError.textContent === "" &&
    usernameInput.value.trim() !== "" && usernameError.textContent === "" &&
    passwordInput.value.trim() !== "" && passwordError.textContent === "" &&
    confirmPasswordInput.value.trim() !== "" && confirmPasswordError.textContent === ""
  );
}
```